RecoLocalCalo/HcalRecProducers/plugins/alpaka/Mahi.dev.cc

```diff
@@ -0,0 +1,1457 @@
+#include <iostream>
+#include <alpaka/alpaka.hpp>
+#include <Eigen/Dense>
+
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/atomicMaxPair.h"
+
+#include "DataFormats/CaloRecHit/interface/MultifitComputations.h"
+// needed to compile with USER_CXXFLAGS="-DCOMPUTE_TDC_TIME"
+#include "DataFormats/HcalRecHit/interface/HcalSpecialTimes.h"
+#include "FWCore/Utilities/interface/CMSUnrollLoop.h"
+// TODO reuse some of the HCAL constats from
+//#include "RecoLocalCalo/HcalRecAlgos/interface/HcalConstants.h"
+
+#include "Mahi.h"
+
+#ifdef HCAL_MAHI_GPUDEBUG
+#define DETID_TO_DEBUG 1125647428
+#endif
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  using namespace cms::alpakatools;
+  namespace hcal::reconstruction {
+    namespace mahi {
+
+      ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE float uint_as_float(unsigned long long int val) {
+#if defined(__CUDA_ARCH__) or defined(__HIP_DEVICE_COMPILE__)
+        return __uint_as_float(val);
+#else
+        return edm::bit_cast<float>(static_cast<unsigned int>(val));
```

Comment:
the `static_cast` can be removed if the type of `val` is changed from `unsigned long long` to `uint32_t`.

Reply:
```suggestion
        return edm::bit_cast<float>(val);
```